verifier/verifier.go

```diff
@@ -518,6 +536,72 @@ func verifyAuthenticTimestamp(outcome *notation.VerificationOutcome) *notation.V
 	}
 }
 
+func verifyRevocation(outcome *notation.VerificationOutcome, client *http.Client, logger log.Logger) *notation.ValidationResult {
+	r, err := revocation.New(client)
+	if err != nil {
+		logger.Debug("error while creating a revocation client, err: %s", err.Error())
+		return &notation.ValidationResult{
+			Type:   trustpolicy.TypeRevocation,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+			Error:  fmt.Errorf("unable to check revocation status, err: %s", err.Error()),
+		}
+	}
+
+	signingTime := outcome.EnvelopeContent.SignerInfo.SignedAttributes.SigningTime
+	revocationResult, err := r.Validate(outcome.EnvelopeContent.SignerInfo.CertificateChain, signingTime)
+	if err != nil {
+		logger.Debug("error while checking revocation status, err: %s", err.Error())
+		return &notation.ValidationResult{
+			Type:   trustpolicy.TypeRevocation,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+			Error:  fmt.Errorf("unable to check revocation status, err: %s", err.Error()),
+		}
+	}
+
+	result := &notation.ValidationResult{
+		Type:   trustpolicy.TypeRevocation,
+		Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+	}
+
+	currResult := revocation_result.ResultUnknown
+	numOKResults := 0
```

Comment:
The purpose of the counter is to ensure that if for some reason the loop ends before visiting a cert with a result of Unknown or Revoked that it will only be considered OK if the number of OK/NonRevokable results was the same as the number of results expected to be parsed. Essentially, it is just further bug protection to avoid false positives